pyclang/runner.py

```diff
@@ -33,7 +33,19 @@ def _get_call_cmd(filepath: str) -> t.List[str]:
         return [realpath] if _is_exe(realpath) else [sys.executable, realpath]
 
     # if s is an executable in $PATH
-    fullpath = shutil.which(filepath)
+    fullpath = ''
+    # On Win function shutil.which() appends all possible extensions from variable PATHEXT. 
+    # Since some commands like `run-clang-tidy` does not have extension, the location failed.
+    # This applies up to python 3.11. In Python 3.12 files without extension can be found.
+    if sys.platform == 'win32' and '.' not in filepath:
+        search_paths = os.environ.get("PATH", "")
+        for directory in search_paths.split(os.pathsep):
```

Comment:
I think `str.split()` might break it in certain cases. Semicolons are valid directory characters on Windows. This will split them up. This is not very probable bug but it is still possible.